### PR TITLE
feat(inputs.exec): Add option to ignore return code

### PR DESCRIPTION
**Pull request by powersj**
## Summary

This allows the exec plugin to ignore any errors returned and continue to parse the output.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

fixes: #9483

### CODE REVIEW

File: plugins/inputs/exec/README.md [hunk 1]
```diff
@@ -34,16 +30,24 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
...
+  # timeout = "5s"
+
+  ## Measurement name suffix
+  ## Used for separating different commands
+  # name_suffix = ""
+
+  ## Ignore Error Code
+  ## If set to true, a non-zero error code in not considered an error and the
+  ## plugin will continue to parse the output.
+  # ignore_error = false
```
Comment (srebhan) [inline plugins/inputs/exec/README.md:42]: Shouldn't we allow the user to specify error codes here? Maybe including a `*` or some special value to signal "all"?

**Review by srebhan — COMMENTED** [review]
The code looks good, but I one question remains...

File: plugins/inputs/exec/README.md (hunk 1 above)
Comment (powersj) [reply plugins/inputs/exec/README.md:42]: That was the direction I originally went with, but then decided if someone wanted to ignore errors, then why make the user research and determine which error codes could come from the command. That could be a long list if they are trying to ignore any error.

Comment (srebhan) [reply plugins/inputs/exec/README.md:42]: I'm fine with accepting as is because I cannot come-up with a proper, simple solution. :-)

**Review by srebhan — APPROVED** [review]

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**